src/windows/users.rs

```diff
@@ -3,31 +3,88 @@
 use crate::sys::utils::to_str;
 use crate::{
     common::{Gid, Uid},
+    windows::sid::Sid,
     User,
 };
 
 use std::ptr::null_mut;
 use winapi::shared::lmcons::{MAX_PREFERRED_LENGTH, NET_API_STATUS};
-use winapi::shared::minwindef::DWORD;
+use winapi::shared::minwindef::{DWORD, LPBYTE};
 use winapi::shared::ntstatus::STATUS_SUCCESS;
 use winapi::shared::winerror::ERROR_MORE_DATA;
-use winapi::um::lmaccess::{NetUserEnum, NetUserGetLocalGroups};
 use winapi::um::lmaccess::{
-    FILTER_NORMAL_ACCOUNT, LG_INCLUDE_INDIRECT, LPLOCALGROUP_USERS_INFO_0, USER_INFO_0,
+    NetUserEnum, NetUserGetInfo, NetUserGetLocalGroups, LOCALGROUP_USERS_INFO_0, USER_INFO_23,
 };
+use winapi::um::lmaccess::{FILTER_NORMAL_ACCOUNT, LG_INCLUDE_INDIRECT, USER_INFO_0};
 use winapi::um::lmapibuf::NetApiBufferFree;
 use winapi::um::ntlsa::{
     LsaEnumerateLogonSessions, LsaFreeReturnBuffer, LsaGetLogonSessionData,
-    PSECURITY_LOGON_SESSION_DATA,
+    SECURITY_LOGON_SESSION_DATA,
 };
-use winapi::um::winnt::{LPWSTR, PLUID};
+use winapi::um::winnt::{LPWSTR, LUID};
 
 // FIXME: once this is mreged in winapi, it can be removed.
 #[allow(non_upper_case_globals)]
 const NERR_Success: NET_API_STATUS = 0;
 
+struct NetApiBuffer<T>(*mut T);
+
+impl<T> Drop for NetApiBuffer<T> {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.0.is_null() {
+                NetApiBufferFree(self.0 as *mut _);
+            }
+        }
+    }
+}
+
+impl<T> Default for NetApiBuffer<T> {
+    fn default() -> Self {
+        Self(null_mut())
+    }
+}
+
+impl<T> NetApiBuffer<T> {
+    pub fn put(&mut self) -> &mut *mut T {
+        assert!(self.0.is_null());
+        &mut self.0
+    }
+
+    pub unsafe fn put_bytes(&mut self) -> &mut LPBYTE {
+        // https://doc.rust-lang.org/std/mem/fn.transmute.html
+        // Turning an &mut T into an &mut U:
+        &mut *(self.put() as *mut *mut T as *mut LPBYTE)
+    }
+}
+
+struct LsaBuffer<T>(*mut T);
+
+impl<T> Drop for LsaBuffer<T> {
+    fn drop(&mut self) {
+        unsafe {
+            if !self.0.is_null() {
+                LsaFreeReturnBuffer(self.0 as *mut _);
+            }
+        }
+    }
+}
+
+impl<T> Default for LsaBuffer<T> {
+    fn default() -> Self {
+        Self(null_mut())
+    }
+}
+
+impl<T> LsaBuffer<T> {
+    pub fn put(&mut self) -> &mut *mut T {
```

Comment:
Just realized but this naming is super weird: you don't put anything anywhere, you just return a mutable reference to your pointer. What do you think about `inner_mut`?